### DISCUSSION TITLE
What is MaxDisplayNumber sesman option?

### DISCUSSION THREAD

**Discussion by Mogaba** [neutrinolabs/xrdp#3298] [Q&A] [answered]
Hello, 

I upgraded xrdp from v0.9.24 to v0.10.1 and found out there's a new sesman option `MaxDisplayNumber`. It seems it affects the maximum number of sessions like `MaxSessions`. The default value 63 doesn't allow more than 54 sessions. After 54 is reached, any attempt to log in results in an error: `X server -- no display in range (10 to 63) is available`.
I couldn't find any info on that except the description in `sesman.ini`:
```
IANA only allocates TCP ports up to 6063 for X servers. If you are not
allowing TCP connections to your X servers you may safely increase this
number.
```
I don't fully understand how it works and if increasing the value is OK. I set it to 110 because I have about 90 users that need to connect to the server. Could someone please explain this option (especially the part about not allowing TCP connections to X server)?

**Comment by matt335672** [comment, marked as answer]
I'll have a go at explaining it!

On Unix/Linux, there are two ways to connect to an X server:-
1) Over a local socket (Unix Domain Socket)
2) Over a TCP socket.

Here's the relevant section from `Xorg(1)`:-

<pre>
NETWORK CONNECTIONS
       Xorg  supports  connections  made  using  the  following reliable byte-
       streams:

       Local
           On most platforms, the "Local" connection  type  is  a  UNIX-domain
           socket.   On  some System V platforms, the "local" connection types
           also include STREAMS pipes, named pipes, and some other mechanisms.
           See the "LOCAL CONNECTIONS" section of X(7) for details.

       TCP/IP
           Xorg listens on port 6000+n, where n is the display  number.   <b>This
           connection  type is usually disabled by default</b>, but may be enabled
           with the -listen option (see the Xserver(1) man page for details).
</pre>

The X server can listen on both network connection types at once, if that's what you want.

If you're not using the `-listen tcp` option on your Xorg instances, you're not using TCP to connect to your X servers, and you can safely increase `MaxDisplayNumber`

If you are using the `-listen tcp` option, the `MaxDisplayNumber` value is important to keep the TCP ports used by the X server in the allowable range of 6000-6063.

Is that clear? I'm happy to provide more information if I've not explained it well enough.

**Reply by Mogaba** [reply]
Thanks for the explanation, I checked my Xorg processes:

`Xorg :73 -auth .Xauthority -config xrdp/xorg.conf -noreset -nolisten tcp -logfile .xorgxrdp.%s.log`

`-nolisten tcp` - this means everything's fine, right?

**Reply by matt335672** [reply]
Definitely.
